### DISCUSSION TITLE
LoRA fine-tuning Evo2-7B: Hitting a wall with layer selection & catastrophic forgetting

### DISCUSSION THREAD

**Discussion by pingdryang-oss** [ArcInstitute/evo2#214] [Q&A]
<html><head></head><body><p>Hi everyone! First off, I absolutely love the work on Evo2 — it's been fascinating to dig into.</p>
<p>I've been experimenting with LoRA fine-tuning Evo2-7B on a domain-specific genomic dataset (~5.6GB) using Savanna, but I've been scratching my head over the results lately and would love a sanity check from the team.</p>
<p><strong>The setup:</strong> I'm using a mix of target data (66.7%) and replay data (33.3%, mostly Evo2 pretraining subset + hg38). My LoRA config is <code>r=32</code>, <code>alpha=64</code> on dense/w modules, targeting layers <code>[24, 25, 26]</code> with an effective LR of ~2e-4.</p>
<p><strong>The weird part:</strong> After tens of thousands of steps, the model barely learns the new domain (train ΔPPL is practically flat at -0.005) but is actively forgetting general genomic representations (hg38 test ΔPPL increased by +0.071). Here are the numbers:</p>
<p>Evaluation uses two metrics: <strong>PPL</strong> (perplexity from next-token prediction loss — lower = better fit) and <strong>Identity</strong> (nucleotide-level accuracy of autoregressive generation: prompt with 3500 tokens → generate 500 tokens, compare to ground truth — higher = better). Deltas are fine-tuned minus base, so negative ΔPPL and positive ΔIdentity indicate improvement.</p>

Data Type | ΔPPL (ft−base) | ΔIdentity (ft−base)
-- | -- | --
Domain-specific (train split) | −0.005 | +0.78%
Domain-specific (test split) | +0.042 | −0.41%
hg38 replay (test split) | +0.071 | −1.45%
Evo2 pretraining replay (test split) | +0.029 | −0.20%

<p>The fine-tuned model barely improves on the target domain (ΔPPL ≈ 0 on train, worse on test) while forgetting general genomic representations (PPL increases across all replay test sets) — even with ~33% replay data in the training mix.</p>
<p>I spent some time staring at the architecture and my data, and I'm super curious about a few things:</p>
<ol>
<li>
<p><strong>Did I break the Hyena trios?</strong> I noticed the repeating <code>[hyena_se, hyena_mr, hyena]</code> trios coupled with <code>flash_v2</code> attention. By targeting layers 24-26, I essentially cut a trio in half (missing the hyena operator at layer 27). Does anyone have an intuition on whether we should be targeting complete functional blocks together? More broadly, are there specific layers that are historically more "LoRA-friendly" for Evo2?</p>
</li>
<li>
<p><strong>Why isn't 33% replay holding the general knowledge?</strong> I'm a bit surprised the model is still experiencing catastrophic forgetting with that much replay data. For those familiar with OpenGenome2, are there specific subsets (like Euk 5kb or GTDB) that are absolutely critical to anchor the general representations? Or is it a matter of tweaking the ratio?</p>
</li>
<li>
<p><strong>Is this just a capacity/LR issue?</strong> Part of me wonders if 3 layers just don't provide enough surface area for Evo2-7B to adapt to a 5.6GB dataset. If I need to scale up, would you instinctively lean towards unfreezing more layers (maybe 21-31) or just cranking up the LoRA rank? Also, does ~2e-4 effective LR feel right for this architecture?</p>
</li>
</ol>
<p>My poor GPU is crying after tens of thousands of steps with no gains. 🥲 If anyone from the core dev team could drop a crumb of internal lore or point me in the right direction, both my GPU and I would be eternally grateful! ☕️✨</p></body></html>
